File: music-commons-api/src/main/java/net/transgressoft/commons/music/playlist/MutableAudioPlaylistDirectory.java
package net.transgressoft.commons.music.playlist;

import net.transgressoft.commons.music.audio.AudioItem;

import java.util.Set;

public interface MutableAudioPlaylistDirectory<I extends AudioItem> extends AudioPlaylistDirectory<I>, MutableAudioPlaylist<I> {

    <N extends AudioPlaylist<I>> void addPlaylists(N... playlists);

    <N extends AudioPlaylist<I>> void addPlaylists(Set<N> playlists);

    <N extends AudioPlaylist<I>> void removePlaylists(N... playlists);

    <N extends AudioPlaylist<I>> void removePlaylists(Set<N> playlists);
}

File: music-commons-api/src/main/kotlin/net/transgressoft/commons/music/playlist/MutableAudioPlaylistDirectory.kt
package net.transgressoft.commons.music.playlist

import net.transgressoft.commons.music.audio.AudioItem

interface MutableAudioPlaylistDirectory<I : AudioItem, N : AudioPlaylist<I>> : AudioPlaylistDirectory<I, N>, MutableAudioPlaylist<I> {

    fun addPlaylists(vararg playlists: N) {
        addPlaylists(setOf(*playlists))
    }

    fun addPlaylists(playlists: Set<N>)

    fun removePlaylists(vararg playlists: N) {
        removePlaylists(setOf(*playlists))
    }

    fun removePlaylists(playlists: Set<N>)
}